source/adapters/level_zero/command_buffer.cpp

```diff
@@ -922,6 +922,13 @@ UR_APIEXPORT ur_result_t UR_APICALL urCommandBufferAppendUSMMemcpyExp(
 
   bool PreferCopyEngine = !IsDevicePointer(CommandBuffer->Context, Src) ||
                           !IsDevicePointer(CommandBuffer->Context, Dst);
+  // For better performance, Copy Engines are not preferred given Shared
+  // pointers on DG2.
+  if ((IsSharedPointer(CommandBuffer->Context, Src) ||
+       IsSharedPointer(CommandBuffer->Context, Dst)) &&
+      CommandBuffer->Device->isDG2()) {
+    PreferCopyEngine = false;
+  }
   PreferCopyEngine |= UseCopyEngineForD2DCopy;
 
```

Comment:
So this "UseCopyEngineForD2DCopy" will not get set if logic for IsSharedPointer, above is set?   Also is the variable "isDG2" also true for ATS-M?

Reply:
UseCopyEngineForD2DCopy is only set from an environment variable. It is a "debug" variable that can be used to override the defaults we are setting above. We need to allow that env to set the value if it is set since it is for "testing" this type of use case for performance measurement. It is default false. 

isDG2 is for all Arc and ATS-M products.

Reply:
Thanks.   Everything looks good.